examples/step-55/step-55.cc

```diff
@@ -0,0 +1,1212 @@
+                                      Point<dim>(-2,0,-1));
+      const Point<dim> top_right   = (dim == 2 ?
+                                      Point<dim>(2,0) :
+                                      Point<dim>(2,1,0));
+
+      GridGenerator::subdivided_hyper_rectangle (triangulation,
+                                                 subdivisions,
+                                                 bottom_left,
+                                                 top_right);
+    }
+
+    for (typename Triangulation<dim>::active_cell_iterator
+         cell = triangulation.begin_active();
+         cell != triangulation.end(); ++cell)
+      for (unsigned int f=0; f<GeometryInfo<dim>::faces_per_cell; ++f)
+        if (cell->face(f)->center()[dim-1] == 0)
+          cell->face(f)->set_all_boundary_ids(1);
+
+
+    triangulation.refine_global (6-dim);
+
+    for (unsigned int refinement_cycle = 0; refinement_cycle<3;
+         ++refinement_cycle)
+      {
+        std::cout << "Refinement cycle " << refinement_cycle << std::endl;
+
+        if (refinement_cycle > 0)
+          triangulation.refine_global (1);
+
+        use_multigrid = false;
+        std::cout << "Now running with ILU" << std::endl;
+        computing_timer.enter_subsection ("Setup");
+        setup_dofs();
+        computing_timer.leave_subsection();
+        std::cout << "   Assembling..." << std::endl << std::flush;
+        computing_timer.enter_subsection ("Assemble");
+        assemble_system ();
+        computing_timer.leave_subsection();
+        std::cout << "   Solving..." << std::flush;
+        computing_timer.enter_subsection ("Solve");
+        solve ();
+        computing_timer.leave_subsection();
+
+        computing_timer.print_summary ();
+        computing_timer.reset ();
+        output_results (refinement_cycle);
+
+        std::cout << std::endl;
+        std::cout << "Now running with Multigrid" << std::endl;
+        use_multigrid = true;
+        computing_timer.enter_subsection ("Setup");
+        setup_dofs();
+        computing_timer.leave_subsection();
+        std::cout << "   Assembling..." << std::endl << std::flush;
+        computing_timer.enter_subsection ("Assemble");
+        assemble_system ();
+        computing_timer.leave_subsection();
+        std::cout << "   Assembling Multigrid..." << std::endl << std::flush;
+        computing_timer.enter_subsection ("Assemble Multigrid");
+        assemble_multigrid ();
+        computing_timer.leave_subsection();
+        std::cout << "   Solving..." << std::flush;
+        computing_timer.enter_subsection ("Solve");
+        solve ();
+        computing_timer.leave_subsection();
+
+        computing_timer.print_summary ();
+        computing_timer.reset ();
+        output_results (refinement_cycle);
+      }
+  }
+}
+
+// @sect3{The main function}
+
+int main ()
+{
+  try
+    {
+      using namespace dealii;
+      using namespace Step22;
+
+      deallog.depth_console (0);
```

Comment:
Doing this creates a lot of output!

i.e.
...
DEAL:GMRES:cg::Starting value 0.00346008
DEAL:GMRES:cg::Convergence step 1 value 8.51862e-19
DEAL:GMRES:cg::Starting value 0.00490014
DEAL:GMRES:cg::Convergence step 1 value 7.22320e-19
DEAL:GMRES:cg::Starting value 0.00173238
DEAL:GMRES:cg::Convergence step 1 value 4.72886e-19
DEAL:GMRES:cg::Starting value 0.00418114
DEAL:GMRES:cg::Convergence step 1 value 5.99200e-19
DEAL:GMRES:cg::Starting value 0.00397920
DEAL:GMRES:cg::Convergence step 1 value 9.54294e-19
DEAL:GMRES:cg::Starting value 0.00314093
DEAL:GMRES:cg::Convergence step 1 value 7.38820e-19
DEAL:GMRES:cg::Starting value 0.00346297
DEAL:GMRES:cg::Convergence step 1 value 4.99117e-19
DEAL:GMRES:cg::Starting value 0.00374311
DEAL:GMRES:cg::Convergence step 1 value 5.54207e-19
DEAL:GMRES:cg::Starting value 0.00592729
DEAL:GMRES:cg::Convergence step 1 value 1.40514e-18
DEAL:GMRES::Convergence step 23 value 1.42760e-05
...